jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1805,7 +1805,7 @@ public void setMetaData(org.eclipse.jetty.http.MetaData.Request request)
         }
         else if (encoded.startsWith("/"))
         {
-            path = (encoded.length() == 1) ? "/" : URIUtil.canonicalPath(URIUtil.decodePath(encoded));
+            path = (encoded.length() == 1) ? "/" : URIUtil.canonicalPath(uri.getDecodedPath());
```

Comment:
oops this is an unrelated change that accidentally slipped in.  
The `uri.getDecodedPath()` method  does exactly `URIUtil.decodePath(encoded)`, but caches the results so they can be reused if needed again.